File: Pandacap.ActivityPub.Communication/IActivityPubCommunicationPrerequisites.cs
namespace Pandacap.ActivityPub.Communication
{
    public interface IActivityPubCommunicationPrerequisites
    {
        string UserAgent { get; }

        Task<string> GetPublicKeyAsync();
        Task<byte[]> SignRsaSha256Async(byte[] data);
    }
}

File: Pandacap.LowLevel/ActivityPub/Communication/IActivityPubCommunicationPrerequisites.fs
namespace Pandacap.ActivityPub.Communication

open System
open System.Threading.Tasks

type IActivityPubCommunicationPrerequisites =
    abstract member UserAgent: string
    abstract GetPublicKeyAsync: unit -> Task<String>
    abstract SignRsaSha256Async: byte[] -> Task<byte[]>
